### DISCUSSION TITLE
Possible to create a cellranger 'filtered_contig_annotations'-like output from '10x-sc-xcr-vdj' outputs?

### DISCUSSION THREAD

**Discussion by jeremymsimon** [milaboratory/mixcr#1430] [Q&A] [answered]
Hi,
I'd like to ultimately use `immunarch` to track and plot clonotype proportions from 10x scTCR-seq data across samples in a _paired-chain_ fashion. Currently this tool can import `MiXCR` outputs, however the paired analysis is only supported for cellranger `filtered_contig_annotations.csv` formatted inputs. 

Is there some way to reliably convert the `clone.groups.tsv` or other outputs here to a cellranger-like format? I know this sort of formatting coercion is possible as I believe `TRUST4` includes a script to create this sort of output.

Thanks!

**Comment by mizraelson** [comment, marked as answer]
Hi, currently we do not provide an automated output in the exact format required by 10x Genomics. However, you can manually create this output by writing a short script, as MiXCR provides information for most of the columns mentioned in the 10x Genomics documentation [here](https://support.10xgenomics.com/single-cell-vdj/software/pipelines/latest/output/annotation#contig).

When transferring data from MiXCR to match the 10x Genomics format, many of the columns can be directly mapped. For the other columns, you can set the following default values:

is_cell: Set to `TRUE` for all rows
high_confidence: Set to `TRUE` for all rows
raw_clonotype_id: Set to `None` for all rows
raw_consensus_id: Set to `None` for all rows

Let me know if you have questions regarding other fields.

**Comment by jeremymsimon** [comment]
Thanks @mizraelson, and if I understand correctly, CellRanger only gives one single V/J/CDR3 per chain. Would the equivalent here to be to take only MiXCR's "primary" calls?

**Reply by mizraelson** [reply]
CellRanger has both outputs with all hits in `all_contig_annotations.csv` and only top hits in `filtered_contig_annotations.csv`. In the [`mixcr exportCloneGroups`](https://mixcr.com/mixcr/reference/mixcr-export/#clone-groups-by-cell) we export the most probable chain for every cell also resolving impossible pairs like TCR and BCR clones in the same cell. This grouped output is available in MiXCR with 10x preset by default.

**Reply by misrak** [reply]
So, what would be the input for immunarch? 
As immunarch want this file "analysis.clonotypes.<chains>.txt" if I am not mistaken this we get from exportClones command?

**Reply by mizraelson** [reply]
Yes, however, it might require some manual modifications, but all the information is stored in the output of the exportClones command.
